### ISSUE TITLE
Cannot empty hasMany related model

### ISSUE THREAD

**Issue by arnaudvalle** [vuex-orm/vuex-orm#487]
Hello, thanks for your great work on this plugin!

I'm facing a problem when I try to update related models.

I have a `User` model defined like so:

```js
export default class User extends Model {
  static entity = 'users';

  static fields() {
    return {
      id: this.increment(),
      skills: this.hasMany(Skill, '$userId'),
    };
  }
...
```

and the related `Skill` model defined like:

```js
export default class Skill extends Model {
  static entity = 'skills';

  static primaryKey = ['$userId', 'id'];

  static fields() {
    return {
      id: this.increment(),
      $userId: this.attr(0),
    };
  }
...
```

At some point in the application, I update the skills of a user like so:

```js
User.update({
  data: data, // <-- this is the data coming from the response of a REST endpoint with the updated skills
  create: ['skills'],
});
```

This works fine as long as I just add / remove one or more skills but it doesn't do anything when `data.skills` is `[]`. The skills entities stay the same.

Am I missing something?

Many thanks,

**Comment by kiaking** [comment]
HI thank you for the report!

Your desired behavior is to have empty Skills after inserting `data.skills` of `[]` right? Hmmm...

I haven't tested this, but maybe it is because the empty field is not present after the data normalization by normalizr. I found this issue so it looks like I'm correct.
https://github.com/paularmstrong/normalizr/issues/290

Hmmm... If that is the reason, not sure how to solve this one... Maybe it's time to build our own normalizer functionality...

**Comment by arnaudvalle** [comment]
Thanks for looking into it. Yes I believe you're right. It seems like there are quite a few issues related to this in the normalizr project and it seems like it's just ignored if the value is falsy :(

As a workaround, I'm deleting the related skills by hand in the success method of my REST call like this:

```
.then(({ data }) => {
  User.update({
    data: data,
    create: ['skills'],
  });
  if (data.skills.length === 0) {
    Skill.delete((skill) => skill.$userId === user.id);
  }
});
```

It's not ideal but does the job for now... 🤷‍♂

**Comment by kiaking** [comment]
Yeah, let me keep this issue open and try to find out some solution for this. I could come up with;

**Implement our own normalizer functionality.** It could be a good idea since there're lots of features in Normalizr where Vuex ORM doesn't use.
**Add some extra option to delete all records.** A bit cumbersome, though it will be more flexible after all. Something like *deleteIfEmpty*.
